apidoc/Titanium/Android/NotificationChannel.yml

```diff
@@ -0,0 +1,82 @@
+---
+name: Titanium.Android.NotificationChannel
+summary: Module for notification channels.
+description: |
+    You can create a notification channel for each distinct type of notification you need to send.
+
+    To create a notification channel, use the <Titanium.Android.createNotificationChannel> method.
+
+    Further Reading:
+
+      * [Android Developers: Notification Channel](https://developer.android.com/reference/android/app/NotificationChannel.html)
+
+examples:
+  - title: Notification Channel Example
+    example: |
+        The following example shows how to create a notification with a channel.
+
+            var channel = Ti.Android.NotificationManager.createNotificationChannel({
+                    id: 'my_channel',
+                    name: 'TEST CHANNEL',
+                    importance: Ti.Android.IMPORTANCE_DEFAULT
+                }),
+                notification = Ti.Android.createNotification({
+                    icon: Ti.Android.R.drawable.ic_dialog_info,
+                    contentTitle: 'TITLE',
+                    contentText : 'This is a test',
+                    channelId: channel.getId()
+                });
+
+            Ti.Android.NotificationManager.notify(100, notification);
+
+extends: Titanium.Proxy
+since: "6.2.0"
+osver: {android: {min: "8.0"}}
+platforms: [android]
+
+properties:
+  - name: bypassDnd
+    summary: Whether or not notifications posted to this channel can interrupt the user.
+    type: Boolean
+
+  - name: description
+    summary: user visible description of this channel.
+    type: String
+
+  - name: enableLights
+    summary: Whether notifications posted to this channel should display notification lights
+    type: Boolean
+
+  - name: enableVibration
+    summary: Whether notification posted to this channel should vibrate.
+    type: Boolean
+
+  - name: groupId
+    summary: Group id this channel belongs to.
+    type: String
+
+  - name: importance
+    summary: The audio stream type to use when playing the sound.
+    description: Specify one of the `IMPORTANCE` constants from <Titanium.Android>.
+    constants: [Titanium.Android.IMPORTANCE_*]
+    type: Number
+
+  - name: id
+    summary: The channel id specified for the notification channel.
+    type: String
+
+  - name: lightColor
+    summary: The notification light color for notifications posted to this channel.
+    type: Number
+
+  - name: lockscreenVisibility
+    summary: Whether or not notifications posted to this channel are shown on the lockscreen in full or redacted form.
+    type: Number
+
+  - name: showBadge
+    summary: Whether notifications posted to this channel can appear as application icon badges in a Launcher.
+    type: Boolean
+
+  - name: vibratePattern
+    summary: The vibration pattern for notifications posted to this channel.
```

Comment:
We should document how this array of numbers work. I believe each array element is a duration in milliseconds, right? Where the even indexed elements (ex: 0, 2, 4, 6, etc.) is the vibration time and the odd indexed elements are how long to wait between vibrations.